crypto/ring-signature/signer/Cargo.toml

```diff
@@ -0,0 +1,39 @@
+[package]
+name = "mc-crypto-ring-signature-signer"
+version = "1.3.0-pre0"
+authors = ["MobileCoin"]
+edition = "2021"
+readme = "README.md"
+
+[dependencies]
+# External dependencies
+displaydoc = { version = "0.2", default-features = false }
+generic-array = { version = "0.14", features = ["serde", "more_lengths"] }
+hex_fmt = "0.3"
+prost = { version = "0.10", default-features = false, features = ["prost-derive"] }
+rand_core = { version = "0.6", default-features = false }
+serde = { version = "1.0", default-features = false, features = ["alloc", "derive"] }
+subtle = { version = "2.4.1", default-features = false, features = ["i128"] }
+zeroize = { version = "1", default-features = false }
+
+# MobileCoin dependencies
+mc-account-keys = { path = "../../../account-keys", default-features = false }
+mc-crypto-keys = { path = "../../keys", default-features = false }
+mc-crypto-ring-signature = { path = "..", default-features = false }
+mc-transaction-types = { path = "../../../transaction/types" }
+mc-util-serial = { path = "../../../util/serial" }
+
+[target.'cfg(any(target_feature = "avx2", target_feature = "avx"))'.dependencies]
+curve25519-dalek = { version = "4.0.0-pre.2", default-features = false, features = ["simd_backend", "nightly"] }
+
+[target.'cfg(not(any(target_feature = "avx2", target_feature = "avx")))'.dependencies]
+curve25519-dalek = { version = "4.0.0-pre.2", default-features = false, features = ["nightly", "u64_backend"] }
```

Comment:
Rather than repeating this gnarly dep, how about we make `mc-crypto-keys` (or similar) re-export some `curve25519_dalek` types? AFAICT we only need `CompressedRistretto`, `RistrettoPoint`, and `Scalar`

Reply:
this sounds good to me, @jcape how do you feel about that?

the thing I think was, James wanted to encourage us to use wrapped types rather than raw dalek types, so he didn't make crypto keys export these things at the beginning. But it's true that, right now we have a lot of code that uses the raw dalek types. 

maybe we can re-export them from a module like `mc_crypto_keys::curve25519_dalek::{Scalar, ...}` so that it is segregated a bit from the other parts of `mc_crypto_keys` public API?

Reply:
Both those approaches SGTM.

Reply:
This crate is going to end up including it's own higher level crypto, using primitives from dalek, so I think a native dep on dalek is fine here.

If we want to avoid having this dep, I suggest asking if the default backend for dalek could be "auto", and use cpuflags...

Reply:
maybe the right thing to do for now is have `mc-crypto-ring-signature` re-export the symbols (which I think it is already doing), and this crate can consume it from there, since it has to match that crate anyways. I don't think it actually needs any of these types except to pass them to the bulletproofs crate at this point.

Reply:
for now just keeping the dep as james suggested

Reply:
I've forked it for our purposes, and we specifically patch crates.io in the `foo-enclave-trusted` crates. When we are using our own custom `x86_64-mobilecoin-none-sgx` JSON target, our fork will be unnecessary.